sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs

```diff
@@ -180,7 +201,16 @@ protected string GetRecordedOptionalVariable(string name)
         /// </summary>
         protected string GetRecordedVariable(string name)
         {
-            var value = GetRecordedOptionalVariable(name);
+            return GetRecordedVariable(name, null);
+        }
+
+        /// <summary>
+        /// Returns and records an environment variable value when running live or recorded value during playback.
+        /// Throws when variable is not found.
+        /// </summary>
+        protected string GetRecordedVariable(string name, Action<RecordedVariableOptions> options)
```

Comment:
Why is this preferable to just using an options bag?

Reply:
sometimes might be shorter `o => o.HasSecretConnectionStringParameter(..)` vs `new RecordedVariableOptions().HasSecretConnectionStringParameter(..)`

Reply:
I mean we only have two cases really that we are supporting, so even shorter would be:
`GetSecretRecordedVariable(string name)`
`GetConnectionStringRecordedVariable(string variableName, string secretName)`

Reply:
Some connection strings get multiple parts replaced with different values.

Reply:
Really?

Reply:
Ah ok I see the IOT usage - there is one secret, and then they also replace the host name.